docs/source/transfer_learning.rst

```diff
@@ -52,16 +52,23 @@ Example: Imagenet (computer Vision)
 
     class ImagenetTransferLearning(LightningModule):
         def __init__(self):
+            super().__init__()
+
             # init a pretrained resnet
-            num_target_classes = 10
-            self.feature_extractor = models.resnet50(pretrained=True)
-            self.feature_extractor.eval()
+            backbone = models.resnet50(pretrained=True)
+            num_filters = backbone.fc.in_features
+            _layers = list(backbone.children())[:-1]
+            self.feature_extractor = torch.nn.Sequential(*_layers)
 
             # use the pretrained model to classify cifar-10 (10 image classes)
-            self.classifier = nn.Linear(2048, num_target_classes)
+            num_target_classes = 10
+            self.classifier = nn.Linear(num_filters, num_target_classes)
 
         def forward(self, x):
-            representations = self.feature_extractor(x)
+            self.feature_extractor.eval()
+            batch_size = x.size(0)
+            with torch.no_grad():
+                representations = self.feature_extractor(x).view(batch_size, -1)
```

Comment:
```suggestion
            with torch.no_grad():
                representations = self.feature_extractor(x).flatten(1)
```
let's use flatten here?

Reply:
perhaps it's not even necessary, as the goal of these docs is probably just to show the pseudo code and get the idea across :)